medicines/web/src/services/documents-loader.ts

```diff
@@ -0,0 +1,36 @@
+import DataLoader from 'dataloader';
+import { IDocument } from '../model/substance';
+
+const graphQlUrl = process.env.GRAPHQL_URL as string;
+
+const fetchFromGraphQl = (query: string): Promise<Response> => {
+  return fetch(graphQlUrl, {
+    method: 'POST',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ query, variables: null }),
+  });
+};
+
+const getDocumentsForProduct = async (productName: any) => {
+  const query = `{ product(name: "${productName}") { documents {       productName
```

Comment:
Probably better to use [a graphQL variable](https://graphql.org/learn/queries/#variables) here so there are no issues if the product name includes a special character like a double quote. It would look something roughly like this I think:

```gql
query ($productName: String!) {
  product(name: $productName) {
    documents {
      productName
    }
  }
}
```

There's also a helper function here now for making graphql requests: https://github.com/MHRA/products/blob/master/medicines/web/src/services/graphql.ts#L8

I think the query might be out of date with the schema now though?

Reply:
used this, looks way better, thank you!